src/bootstrap/compile.rs

```diff
@@ -220,15 +220,19 @@ pub fn std_cargo(builder: &Builder<'_>,
             .arg("--manifest-path")
             .arg(builder.src.join("src/libtest/Cargo.toml"));
 
+        // Help the libc crate compile by assisting it in finding various
+        // sysroot native libraries.
         if target.contains("musl") {
             if let Some(p) = builder.musl_root(target) {
-                cargo.env("MUSL_ROOT", p);
+                let root = format!("native={}/lib", p.to_str().unwrap());
+                cargo.rustflag("-L").rustflag(&root);
```

Comment:
Hm, don't we need to do this for libc independent of what depends on it (e.g., compiler, cargo, etc?), not just for libc-as-a-std-dep?

Reply:
No this is only for the libstd build where the C library is linked in statically, so it's only needed for the std build.